pennylane/pulse/parametrized_hamiltonian_pytree.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module containing the ``JaxParametrizedHamiltonian`` class."""
+from dataclasses import dataclass
+from typing import Callable, Tuple, Union
+
+import pennylane as qml
+
+from .parametrized_hamiltonian import ParametrizedHamiltonian
+
+has_jax = True
+try:
+    import jax
+    import jax.numpy as jnp
+    from jax.experimental import sparse
+    from jax.tree_util import register_pytree_node_class
+except ImportError:
+    has_jax = False
+
+
+@register_pytree_node_class
+@dataclass
+class ParametrizedHamiltonianPytree:
+    """Jax pytree class that represents a ``ParametrizedHamiltonian``."""
+
+    mat_fixed: Union[jnp.ndarray, sparse.CSR]
+    mats_parametrized: Tuple[Union[jnp.ndarray, sparse.CSR], ...]
+    coeffs_parametrized: Tuple[Callable]
+
+    @staticmethod
+    def from_hamiltonian(H: ParametrizedHamiltonian, dense: bool = False, wire_order=None):
```

Comment:
can we enforce those to be kwargs?
```suggestion
    def from_hamiltonian(H: ParametrizedHamiltonian, *, dense: bool = False, wire_order=None):
```

Reply:
Why do we want to enforce this? Adding `*` is dangerous, because we are ignoring all args after `H`:
```python
# True is ignored
>>> ParametrizedHamiltonianPytree.from_hamiltonian(H, True)
```

Reply:
Are you sure? 
I just tried and it errors
```python
In [2]: def test(x, *, arg1=False, arg2=False):
   ...:     print(f"{x=},{arg1=},{arg2=}")
   ...:

In [3]: test(1, 2)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [3], in <cell line: 1>()
----> 1 test(1, 2)

TypeError: test() takes 1 positional argument but 2 were given
```

what `*` does (as opposed to `*args`) is simply that it enforces the following arguments to be specified with keywords.

Reply:
You're right 😮 I didn't know this! Then I have no problem in adding this 👍